druid/src/widget/widget.rs

```diff
@@ -46,6 +46,11 @@ use super::prelude::*;
 /// [`LifeCycleCtx::widget_id`]: struct.LifeCycleCtx.html#method.widget_id
 /// [`WidgetExt::with_id`]: trait.WidgetExt.html#method.with_id
 /// [`IdentityWrapper`]: widget/struct.IdentityWrapper.html
+//
+// The widget id does not necessarily identify only a single widget
+// in the strictest sense. Internally WidgetPod uses its child's id,
```

Comment:
Kind of. In the general case, the `WidgetPod` is the source of the id; it only uses the child's id if that child has been given an explicit id via `IdentityWrapper`, which is only done when the user wants to know the id ahead of time.

So it's more like: The `WidgetId` will be the same for all `Widget`s that share the same nearest ancestor `WidgetPod`.